File: cdm/cpp/patient/actions/SEConsciousRespirationCommand.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
#include "patient/actions/SEConsciousRespirationCommand.h"

SEConsciousRespirationCommand::SEConsciousRespirationCommand()
{
  m_Comment = "";
}

SEConsciousRespirationCommand::~SEConsciousRespirationCommand()
{
  Clear();
}

void SEConsciousRespirationCommand::Clear()
{
  m_Comment = "";
}

bool SEConsciousRespirationCommand::IsValid() const
{
  return true;
}

bool SEConsciousRespirationCommand::IsActive() const
{
  return true;
}

std::string SEConsciousRespirationCommand::GetComment() const
{
  return m_Comment;
}
void SEConsciousRespirationCommand::SetComment(const std::string& comment)
{
  m_Comment = comment;
}
bool SEConsciousRespirationCommand::HasComment() const
{
  return m_Comment.empty() ? false : true;
}
void SEConsciousRespirationCommand::InvalidateComment()
{
  m_Comment = "";
}

File: cdm/java/mil/tatrc/physiology/datamodel/patient/actions/SEConsciousRespirationCommand.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package mil.tatrc.physiology.datamodel.patient.actions;

public abstract class SEConsciousRespirationCommand
{
  protected String comment;
  
  public SEConsciousRespirationCommand()
  {
  }
  
  public void copy(SEConsciousRespirationCommand other)
  {
    if(this==other)
      return;
    
    comment = other.comment;
  }
  
  public abstract SEConsciousRespirationCommand getCopy();
  
  public void reset()
  {
    comment = "";
  }
  /*
  public static void load(ConsciousRespirationCommandData in)
  {
    comment = in.getComment();
  }
  protected void unload(ConsciousRespirationCommandData data)
  {
    if (hasComment())
      data.setComment(comment);
  }
  */
  public boolean hasComment()
  {
    return comment != null && !comment.isEmpty();
  }
  public String getComment()
  {
    return comment;
  }
  public void setComment(String s)
  {
    comment = s;
  }
  
  public abstract boolean isValid();
  public abstract String toString();
}
